shu-etal.ipynb

```diff
@@ -0,0 +1,1516 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stable = stable1.append(stable2)\n",
+    "progr = progr1.append(progr2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a347dd1",
+   "metadata": {},
+   "source": [
+    "## Sanity checks\n",
+    "\n",
+    "Ensure that the cohort built does not have any defects (duplication, size, etc..)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "6fb365ea",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "All good!\n"
+     ]
+    }
+   ],
+   "source": [
+    "for (a, b) in [(stable1, progr1), (stable2, progr2)]:\n",
+    "    # Both groups have equal size\n",
+    "    assert(len(a) == len(b))\n",
+    "    # No patient is present more than once in each group\n",
+    "    for x in [a, b]:\n",
+    "        patnos_x = pd.unique(x['PATNO'])\n",
+    "        assert(len(patnos_x)==len(x)), x\n",
+    "    # Patients in a aren't in b\n",
+    "    patnos_a = pd.unique(a['PATNO'])\n",
+    "    patnos_b = pd.unique(b['PATNO'])\n",
+    "    for p in patnos_a:\n",
+    "        assert(p not in patnos_b), f'PATNO {p} appears in a and b'\n",
+    "        \n",
+    "# Patients in each cohort do not replicate\n",
+    "assert len(set(pd.unique(progr[\"PATNO\"])) & (set(pd.unique(stable[\"PATNO\"]))))==0, \"Patients occur in both groups\"\n",
+    "\n",
+    "print('All good!')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c40ef320",
+   "metadata": {},
+   "source": [
+    "## Cohort summary"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "1270013e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\t\t\t\t\t| Stable set (n=72)\t| Progression set (n=72)\n",
+      "Age (years) \t\t\t\t| 60.94 \t\t| 60.99\n",
+      "Gender [male (n)] \t\t\t| 46 \t\t\t| 46\n",
+      "Hoehn & Yahr (n) \t Stage 1 \t| 29 \t\t\t| 29\n",
+      "\t\t\t Stage 2 \t| 43 \t\t\t| 43\n",
+      "\t\t\t Stage 3 \t| 0 \t\t\t| 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"\\t\\t\\t\\t\\t| Stable set (n={len(stable)})\\t| Progression set (n={len(progr)})\" + os.linesep +\n",
+    "       f\"Age (years) \\t\\t\\t\\t| {round(stable['AGE_AT_VISIT'].mean(),2)} \\t\\t| {round(progr['AGE_AT_VISIT'].mean(),2)}\" + os.linesep + \n",
+    "       f\"Gender [male (n)] \\t\\t\\t| {len(stable[stable['SEX']==1])} \\t\\t\\t| {len(progr[progr['SEX']==1])}\" + os.linesep +\n",
+    "       f\"Hoehn & Yahr (n) \\t Stage 1 \\t| {len(stable[stable['NHY_x']=='1'])} \\t\\t\\t| {len(progr[progr['NHY_x']=='1'])}\" + os.linesep +\n",
+    "       f\"\\t\\t\\t Stage 2 \\t| {len(stable[stable['NHY_x']=='2'])} \\t\\t\\t| {len(progr[progr['NHY_x']=='2'])}\" + os.linesep +\n",
+    "       f\"\\t\\t\\t Stage 3 \\t| {len(stable[stable['NHY_x']=='3'])} \\t\\t\\t| {len(progr[progr['NHY_x']=='3'])}\"\n",
```

Comment:
Improve table formatting by putting the numbers in a dataframe. See cell 8 in https://github.com/LivingPark-MRI/scherfler-etal/blob/main/scherfler-etal.ipynb